### ISSUE TITLE
settings not WAI in package sub-directory?

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2512]
In https://github.com/Rdatatable/data.table/pull/5908/commits/5e36615783f846964418085d8dcf55f16e7c90f2, I tried to put the `.lintr` config into a non-standard location and point to it in CI with the `env` approach mentioned in `?read_settings`: `R_LINTR_LINTER_FILE: .ci/.lintr`. Also tried with `.ci/.lintr.R` (the exact file, not relying on auto-guessing the R extension).

However, this failed to get picked up, as seen by the 1000s of lints picked up in the associated job: https://github.com/Rdatatable/data.table/actions/runs/7523183440/job/20476211212?pr=5908

Am I doing something wrong?

I've also tried running `options(lintr.linter_file = ".ci/.lintr")` (and `.ci/.lintr.R`) to no avail. I've also print-debugged that `file.exists(".ci/.lintr.R")`, so the file is not in the wrong location / omitted for being in a hidden directory.

Currently, I'm needing to run `lintr:::read_settings(".ci/lintr"); lint_package(parse_settings = FALSE)` to accomplish what I intended. Really not pretty.

PS it might help for `verbose=TRUE` to report some info in `read_settings()` as well, e.g. which file was chosen, maybe even which settings are read as which values.

**Comment by AshesITR** [comment]
Any reason for using `basename(...)` here?

```r
find_local_config(path, basename(linter_file))
```

**Comment by MichaelChirico** [comment]
I assume you're referring to:

https://github.com/r-lib/lintr/blob/bad1632f23b77f41cec13fb6f1c883302f64f506/R/settings_utils.R#L78

IINM that entered the package long ago before the current form, then survived in #2205 without being re-examined:

https://github.com/r-lib/lintr/commit/1a7114b87c4d672772c74ffc30c932dcdb3ce017

But in short I don't know for sure what the aim is there...

**Comment by AshesITR** [comment]
IINM removing the call to `basename()` should fix this issue.
